packages/paste-website/src/pages/components/editable-code-block/index.mdx

```diff
@@ -0,0 +1,370 @@
+  code={`<EditableCodeBlockWrapper>
+  <EditableCodeBlockHeader>ReadOnly Example</EditableCodeBlockHeader>
+  <EditableCodeBlock height="45vh" readOnly defaultLanguage="typescript" defaultValue={TypeScriptExample} />
+</EditableCodeBlockWrapper>`}
+/>
+
+### Syntax errors
+
+Monaco Editor will detect and show syntax errors out of the box.
+
+<StoryPreview
+  title="Syntax error example in Editable Code Block"
+  storyID="components-editable-code-block--syntax-error"
+  code={`<EditableCodeBlockWrapper>
+  <EditableCodeBlockHeader>Syntax error example</EditableCodeBlockHeader>
+  <EditableCodeBlock
+    height="45vh"
+    defaultLanguage="typescript"
+    defaultValue={CodeStringWithSyntaxError}
+  />
+</EditableCodeBlockWrapper>`}
+/>
+
+### Line-specific errors
+
+Show line-specific errors to help users pinpoint what line of code they need to fix.
+
+<StoryPreview
+  title="Custom line-specific errors in Editable Code Block"
+  storyID="components-editable-code-block--custom-inline-error"
+  code={`<EditableCodeBlockWrapper>
+  <EditableCodeBlockHeader>Custom inline error example (try hovering it)</EditableCodeBlockHeader>
+  <EditableCodeBlock
+    height="45vh"
+    onChange={(value) => console.log(value)}
+    inlineErrorRange={{
+      startLineNumber: 3,
+      endLineNumber: 3,
+      startColumn: 7,
+      endColumn: 13,
+    }}
+    inlineErrorIsWholeLine={false}
+    inlineErrorHoverMessage={{value: '"id" can only be a string type.'}}
+    defaultLanguage="typescript"
+    defaultValue={TypeScriptExample}
+  />
+</EditableCodeBlockWrapper>`}
+/>
+
+### Generic errors
+
+If there’s an error that can’t be traced to specific line numbers, use a [Callout](/components/callout).
+
+For additional guidance on how to compose error messages, refer to the [error state pattern](/patterns/error-state).
+
+<StoryPreview
+  title="Generic errors in Editable Code Block"
+  storyID="components-editable-code-block--generic-error"
+  code={`<Stack orientation="vertical" spacing="space40">
+  <Callout variant="error">
+    <CalloutHeading as="h3">401 Unauthorized Request</CalloutHeading>
+    <CalloutText>Please provide a valid API key.</CalloutText>
+  </Callout>
+  <EditableCodeBlockWrapper>
+    <EditableCodeBlockHeader>Generic error example</EditableCodeBlockHeader>
+    <EditableCodeBlock height="45vh" defaultLanguage="typescript" defaultValue="const API_KEY = null;" />
+  </EditableCodeBlockWrapper>
+</Stack>`}
+/>
+
+## When to use Editable Code Block
+
+<DoDont>
+  <Do title="Do" body="Use Editable Code Block for a multi-line code input." />
+  <Dont
+    title="Don't"
+    body="Don’t use Textarea for a multi-line code input. The monospace formatting and line numbers of Editable Code Block can help a user check that their code is entered correctly."
+  />
+</DoDont>
+
+<DoDont>
+  <Do
+    title="Do"
+    body="Pair Editable Code Block with a visible label. This can be the Editable Code Block Header, a page Heading, or Label."
```

Comment:
```suggestion
    body="Pair Editable Code Block with a visible label. This can be the Editable Code Block Header, an associated page Heading, or Label."
```